efi/generate_binary.py

```diff
@@ -73,8 +73,19 @@ def _add_nx_pefile(args):
     try:
         import pefile
     except ImportError:
-        print("Unable to add NX support to binaries without genpeimg or python3-pefile")
-        sys.exit(1)
+        print("Unable to add NX support to binaries without genpeimg or python3-pefile, adding NX support manually")
+        import os
+        with open(args.outfile, 'r+b') as fh:
+                buf = bytearray(fh.read(os.path.getsize(args.outfile)))
+                fh.seek(0)
+                # Set bit 4 (IMAGE_DLLCHARACTERISTICS_NX_COMPAT) of DllCharacteristics within the COFF optional header
+                # DllCharacteristics offset = 0xde size = 2 bytes
+                # https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#optional-header-image-only
+                buf[0xdf] |= 0x1
+                fh.write(buf)
+                fh.close()
```

Comment:
This isn't really needed when you're doing `with` to open is it?

Reply:
Correct, close isn't needed.

Reply:
Ok I wan't sure, will remove it.